### PR TITLE
mcs: support region label http interface in scheduling server

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5839

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 1]
```diff
@@ -955,3 +969,144 @@ func getPlacementRuleByGroup(c *gin.Context) {
...
+		return
+	}
+	labels := l.GetRegionLabels(region)
+	c.IndentedJSON(http.StatusOK, labels)
+}
+
+// @Tags     region_label
+// @Summary  List all label rules of cluster.
+// @Produce  json
+// @Success  200  {array}  labeler.LabelRule
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:1044]: need to update

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 2]
```diff
@@ -955,3 +969,144 @@ func getPlacementRuleByGroup(c *gin.Context) {
...
+	}
+	c.IndentedJSON(http.StatusOK, rules)
+}
+
+// @Tags     region_label
+// @Summary  Get label rule of cluster by id.
+// @Param    id  path  string  true  "Rule Id"
+// @Produce  json
+// @Success  200  {object}  labeler.LabelRule
+// @Failure  404  {string}  string  "The rule does not exist."
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:1090]: ditto

File: pkg/schedule/handler/handler.go [hunk 3]
```diff
@@ -1063,6 +1064,24 @@ func (h *Handler) GetHotBuckets(regionIDs ...uint64) (HotBucketsResponse, error)
 	return ret, nil
 }
 
+// GetRegion returns the region labeler.
+func (h *Handler) GetRegion(id uint64) *core.RegionInfo {
+	c := h.GetCluster()
+	if c == nil {
+		return nil
```
Comment (rleungx) [inline pkg/schedule/handler/handler.go:1071]: return nil?

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 4]
```diff
@@ -167,6 +169,13 @@ func (s *Service) RegisterOperatorsRouter() {
 	router.GET("/records", getOperatorRecords)
 }
 
+// RegisterRegionsRouter registers the router of the regions handler.
+func (s *Service) RegisterRegionsRouter() {
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:173]: Shall we move it to config router?

Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go:173]: ok, how about "/config/region/:id/label/:key" and "/config/region/:id/labels"

File: pkg/schedule/handler/handler.go (hunk 3 above)
Comment (lhy1024) [reply pkg/schedule/handler/handler.go:1071]: Do you have any good ideas?

Comment (rleungx) [reply pkg/schedule/handler/handler.go:1071]: Why not return an error

File: pkg/mcs/scheduling/server/apis/v1/api.go (hunk 4 above)
Comment (rleungx) [reply pkg/mcs/scheduling/server/apis/v1/api.go:173]: I think ok until we find a better one.

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 5]
```diff
@@ -191,6 +200,11 @@ func (s *Service) RegisterConfigRouter() {
 	placementRule := router.Group("placement-rule")
 	placementRule.GET("", getPlacementRules)
 	placementRule.GET("/:group", getPlacementRuleByGroup)
+
+	regionLabel := router.Group("region-label")
+	regionLabel.GET("rules", getAllRegionLabelRules)
+	regionLabel.GET("rules/ids", getRegionLabelRulesByIDs)
+	regionLabel.GET("rule/:id", getRegionLabelRuleByID)
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:207]: no need escape anymore?

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 6]
```diff
@@ -991,7 +987,11 @@ func getRegionLabelByKey(c *gin.Context) {
 		return
 	}
 
-	region := handler.GetRegion(id)
+	region, err := handler.GetRegion(id)
+	if err != nil {
+		c.String(http.StatusBadRequest, err.Error())
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:994]: Why 404?

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**